### PR TITLE
chore(deps): update mathjax (major)

### PR DESCRIPTION
**Pull request by ggrossetie**
This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [mathjax](https://redirect.github.com/mathjax/MathJax) | devDependencies | major | [`^2.7.9` → `^4.0.0`](https://renovatebot.com/diffs/npm/mathjax/2.7.9/4.1.2) |

---

### Release Notes

<details>
<summary>mathjax/MathJax (mathjax)</summary>

### [`v4.1.2`](https://redirect.github.com/mathjax/MathJax/releases/tag/4.1.2): MathJax v4.1.2

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/4.1.1...4.1.2)

This is mainly a bug fix release that resolves a number of issues from v4.1.1.

The one change that may affect current content is that some of the font-sizing macros like `\large` and `\tiny` were setting the wrong sizes, and have been corrected.  In particular, the sizes for `\tiny` and `\Tiny` were reversed, while the size for `\large` and the other larger sizes were all off by one size (`\large` was using the size that should have been for `\Large`, `\Large` was using the size of `\LARGE`, etc.), and `\small` was too small.  These have all been fixed to use the correct sizes.  Existing content that use these macros will now use the correct sizing values, and that may affect the results of those expressions.  For this reason, we have added a new `fontsizev3` TeX package that puts back the incorrect sizing values used in v3 and earlier versions of v4.  If you include

```js
window.MathJax = {
  loader: {load: ['[tex]/fontsizev3']},
  tex: {packages: {'[+]': ['fontsizev3']}},
};
```

in your configuration, you will keep the original sizes.  In addition to these size corrections, the missing `\SMALL`, `\Small`, and `\footnotesize` were added, as was a non-standard `\HUGE` to correspond with the size available (incorrectly) as `\Huge` in the past.

See the [release notes for 4.1.2](https://redirect.github.com/mathjax/MathJax-src/releases/tag/4.1.2) for additional details.

### [`v4.1.1`](https://redirect.github.com/mathjax/MathJax/releases/tag/4.1.1): MathJax v4.1.1

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/4.1.0...4.1.1)

This is mainly a bug fix release that resolves a number of issues from v4.1.0. The key fixes include:

- Fixed CS for positioning dialogs so that they appear at the proper location in a scrolled window, and so that page doesn't scroll to the top.
- A new `ui/no-dark-mode` extension has been added to disable MathJax's dark-mode color scheme for those pages that don't implement a dark-mode color set.
- Handling of clicks in form elements embedded within an expression are now better handled by the expression explorer, and tabbing will include embedded form elements.
- The expression explorer now passes mouse clicks and other events on to the surrounding parent tags, so that math within a `<button>` element, for example, will allow the button to receive the click event when you click on the math.
- A number of speech improvements have been made, including better handling of phantom elements, better speech for over/under constructions, and better handling of the speech for tables.
- Speech is now properly added to inline elements in SVG output when automatic in-line line-breaks are allowed.

See the [v4.1.1 release notes](https://redirect.github.com/mathjax/MathJax-src/releases/tag/4.1.1) for additional details.

### [`v4.1.0`](https://redirect.github.com/mathjax/MathJax/releases/tag/4.1.0): MathJax v4.1.0

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/4.0.0...4.1.0)

This release adds a number of new or improved features to MathJax, and fixes several issues in version 4.0.0.  These are described in detail in the [MathJax v4.1.0 release notes](https://redirect.github.com/mathjax/MathJax-src/releases/tag/4.1.0), but among the more important are:

- Changes to make v4.1 more compatible with older browsers (v4.0 used modern CSS and javascript functions that are not available in some older browsers).
- Support for a dark-mode color scheme.
- Movable and sizable dialog boxes (both via the mouse and by keyboard arrow keys).
- Ability to tab to hyperlinks within typeset expressions.
- More accurate highlighting of some expressions during exploration.
- Better handling of tactile Braille output devices in Windows.
- Better handling of speech for tables and tagged expressions.
- Better voicing of European-style open interval notation.
- Fixed arrows in `mhchem` that were swapped accidentally.
- More reliable `data-latex` attributes for TeX output.
- Better positioning of some accents (e.g., the vector arrow) in Safari and other WebKit-based browsers in CHTML output.
- Several new configuration options, e.g., to make font loading easier.
- Improvements to the build system to allow MathJax to be built in Windows power shell.

### [`v4.0.0`](https://redirect.github.com/mathjax/MathJax/releases/tag/4.0.0): MathJax v4.0.0

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.2.2...4.0.0)

This is the long-awaited 4.0.0 release of MathJax, which has been several years in the making.  Among the many new features in this release are:

- Extended font support (11 new fonts)
- A new default font with much greater character coverage.
- Line-breaking support (for in-line as well as displayed equations)
- An updated expression explorer that is now on by default.
- Separation of speech generation into a webworker thread for better performance.
- Support for HTML embedded in TeX and MathML expressions.
- Availability as ES6 modules as well as CommonJS modules.
- New TeX extensions, including the `begingroup` extension from v2 (with more features), and several font-related extensions.
- Inclusion of the `textmacros` extension in all combined component files.
- Updated `mathtools` package to include the changes from 2022 and 2024.
- Improvements in the baseline alignment of text in CHTML output in WebKit-based browsers.
- Improved stretchy delimiters in CHTML output.
- More promise-based conversion and typesetting calls at the document level.

For many sites, the only change that will be needed to use v4 is to change the URL where you load MathJax to refer to version 4 rather than version 3, and to remove the /es5 directory from the URL.  Most v3 configurations should still work with v4; but if you use the startup.ready() function to modify MathJax or create new TeX extensions, there may be some adjustments needed.  One important change of note is that, since the number of TeX packages has grown, the all-packages extension and the "-full" components have been removed, so if you are using those, you will need to change to a component without the "-full" and either rely on the autoload extension to load the needed packages, or load them explicitly.

The [MathJax documentation](https://docs.mathjax.org/en/latest/index.html) has been substantially updated for version 4, and now includes the missing or limited sections on using MathJax in node, including many examples and details.  See the [What's New in V4.0](https://docs.mathjax.org/en/latest/upgrading/whats-new-4.0.html) section (more than 50 pages long) for much more information about the features described above, and in  particular, check the [Breaking Changes](https://docs.mathjax.org/en/latest/upgrading/whats-new-4.0/breaking.html) section for any changes that may affect your usage.

The [MathJax web demos](https://redirect.github.com/mathjax/MathJax-demos-web) repository has been updated for v4.0, and reorganized for easier viewing.  The demos now include a "Show Source" button that displays explanations and the important sections of the source code rather than using separate `.md` files, so there are fewer files to wade through, and you can compare the source code to the output more easily.  There are also new examples in the documentation on using MathJax in a browser; most are listed in the [Examples in a Browser](https://docs.mathjax.org/en/latest/web/examples.html) section.

The [MathJax node demos](https://redirect.github.com/mathjax/MathJax-demos-node) repository is being updated and reorganized as well, but work is still being done on that, as some of the documentation needs to be expanded (for now, you may need to read the comments in the code).  The tools in this repository have been made more powerful and complete, and much of the functionality that is common across multiple tools have been moved to utility libraries that can be shared by several tools.  That means the examples are no longer stand-alone files, but they are more complete in their ability to use the features of MathJax.  For example, all the examples that produce CHTML or SVG output can select the font to use, or add speech labels, and so on, and because speech is now available in all those examples, there are no longer separate speech demos.  The tools can now take several expressions at once to typeset, or take them from a file, or process a complete HTML page.  This means there are no longer separate page-based tools, so the number of examples is reduced, but the functionality is expanded.  The MathJax documentation now covers using MathJax in node application in more detail than for v3, with lots of examples; most of these are listed in the [Examples in Node Applications](https://docs.mathjax.org/en/latest/server/examples.html) section.

As always, you may report issues in the [MathJax issue tracker](https://redirect.github.com/mathjax/MathJax/issues) for v4; this is a major update, and there are sure to be things that will need adjusting.

### [`v3.2.2`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.2.2): MathJax v3.2.2

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.2.1...3.2.2)

This is a hot-fix release that resolves three issues introduced in v3.2.1. See the [release notes](https://redirect.github.com/mathjax/MathJax-src/releases/tag/3.2.2) for details.

### [`v3.2.1`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.2.1): MathJax v3.2.1

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.2.0...3.2.1)

This is a bug fix release that resolves more than 40 issues, and includes significant updates to the speech-rule engine that handles MathJax's assistive technology. See the [release notes](https://redirect.github.com/mathjax/MathJax-src/releases/tag/3.2.1) for details.

### [`v3.2.0`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.2.0): MathJax v3.2.0

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.1.4...3.2.0)

This is a feature release that includes a new lazy-typesetting extension, nine new TeX extensions, a port of the v2 MML3 extension, new Hindi support in the expression explorer, along with several other improvements, and a number of bug fixes.  See the [release notes](https://redirect.github.com/mathjax/MathJax-src/releases/tag/3.2.0) for details.

### [`v3.1.4`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.1.4): MathJax v3.1.4

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.1.3...3.1.4)

This is a hot fix release to fix two problems with the recent 3.1.3 version.  See the [release notes](https://redirect.github.com/mathjax/MathJax-src/releases/tag/3.1.4) for details.

### [`v3.1.3`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.1.3): MathJax 3.1.3

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.1.2...3.1.3)

This is a bug fix release that resolves more than 70 issues, and updates a number of modules on which MathJax relies. See the [release notes](https://redirect.github.com/mathjax/MathJax-src/releases/tag/3.1.3) for details.

### [`v3.1.2`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.1.2): MathJax v3.1.2

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.1.1...3.1.2)

Hot fix to correct regression introduced in v3.1.1 that caused control sequences followed by spaces to not be recognized.

### [`v3.1.1`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.1.1): MathJax v3.1.1

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.1.0...3.1.1)

This is a bug fix release to address issues that arose from the recent 3.1.0 release.  See the [release notes](https://redirect.github.com/mathjax/MathJax-src/releases/tag/3.1.0) for details.

### [`v3.1.0`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.1.0): MathJax v3.1.0

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.0.5...3.1.0)

This is a feature release that includes a number of API improvements, new extensions, and updates to the assistive tools, as described in the [release notes](https://redirect.github.com/mathjax/MathJax-src/releases/tag/3.1.0).

### [`v3.0.5`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.0.5): MathJax v3.0.5

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.0.4...3.0.5)

This is a hot fix release to resolve an important issue with horizontal stretchy characters (e.g., `\overline` and `\overrightarrow`) in CommonHTML output that was introduced in the recent version 3.0.4 release.

### [`v3.0.4`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.0.4): MathJax v3.0.4

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.0.1...3.0.4)

This is a bug-fix release.  See the [release notes](https://redirect.github.com/mathjax/MathJax-src/releases/tag/3.0.4) for details.

Note that version 3.0.2 and 3.0.3 had problems with their `npm` deployment (some files where missing from the package), so were removed.  This is the next actual version after 3.0.1

### [`v3.0.1`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.0.1): MathJax v3.0.1

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/3.0.0...3.0.1)

This is a bug-fix release.  See the [release notes](https://redirect.github.com/mathjax/MathJax-src/releases/tag/3.0.1) for details.

### [`v3.0.0`](https://redirect.github.com/mathjax/MathJax/releases/tag/3.0.0): MathJax v3.0.0

[Compare Source](https://redirect.github.com/mathjax/MathJax/compare/2.7.9...3.0.0)

After over two years of development, the MathJax team is pleased to announce the official release of MathJax 3.0.0!

This version is a complete rewrite of MathJax from the ground up using modern programming techniques, including Typescript, ES6 modules, Promises, and more.

Please see the [What's New in v3.0](http://docs.mathjax.org/en/v3.0-latest/upgrading/whats-new-3.0.html) section of the documentation for details about the new version 3.0.0.

Please see the [Upgrading from v2](http://docs.mathjax.org/en/v3.0-latest/upgrading/v2.html) instructions for details about what is and isn't in version 3, and how to upgrade your site from version 2.  Note in particular that there is a [configuration converter](https://mathjax.github.io/MathJax-demos-web/convert-configuration/convert-configuration.html) that can help you move from version 2 to version 3.

See the [documentation](http://docs.mathjax.org/) for details about how to use MathJax.

See the [MathJax Source Repository](https://redirect.github.com/mathjax/MathJax-src) for the source code for versions 3.

There are two repositories that contain examples of using MathJax, one for web pages, and one for use with node on a server:

- [MathJax Web Examples](https://redirect.github.com/mathjax/MathJax-demos-web#mathjax-demos-web)
- [MathJax Node Examples](https://redirect.github.com/mathjax/MathJax-demos-node#mathjax-demos-node)

Please use the [MathJax Issue Tracker](https://redirect.github.com/mathjax/MathJax/issues) to report problems in version 3 of MathJax.

</details>

---

### Configuration

📅 **Schedule**: (UTC)

- Branch creation
  - At any time (no schedule defined)
- Automerge
  - At any time (no schedule defined)

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get [config help](https://redirect.github.com/renovatebot/renovate/discussions) if that's undesired.

---

 - [ ] If you want to rebase/retry this PR, check this box

---

This PR has been generated by [Mend Renovate](https://redirect.github.com/renovatebot/renovate).

### CODE REVIEW

**Comment by ggrossetie** [comment]
### Edited/Blocked Notification

Renovate will not automatically rebase this PR, because it does not recognize the last commit author and assumes somebody else may have edited the PR.

You can manually request rebase by checking the rebase/retry box above.

 ⚠️ **Warning**: custom changes will be lost.
